iceoryx2-bb/log/src/lib.rs

```diff
@@ -179,6 +186,20 @@ pub enum LogLevel {
     Fatal = 5,
 }
 
+impl LogLevel {
+    fn from_str(s: &str) -> LogLevel {
+        match s {
+            "Trace" => LogLevel::Trace,
+            "Debug" => LogLevel::Debug,
+            "Info" => LogLevel::Info,
+            "Warn" => LogLevel::Warn,
+            "Error" => LogLevel::Error,
+            "Fatal" => LogLevel::Fatal,
+            _ => LogLevel::Info,
+        }
+    }
```

Comment:
What do you think about making this a bit more relaxed by normalizing the input string to lower case. In this case the function name would be more like `from_str_fuzzy`.

I think it would be good to use `printf` and tell the user that the env variable contains an invalid string.

Finally, you can use `DEFAULT_LOG_LEVEL` for the default match arm.

Reply:
I had to change the return from the function to `u8` instead of `LogLevle` and updated the code accordingly

Reply:
Oh, you can also change the type of `DEFAULT_LOG_LEVEL` and `ENV_LOG_LEVEL` to `LogLevel`. Then there is only one `as u8` required :)